tests/tools/oesign/sign-and-verify.py

```diff
@@ -25,21 +35,18 @@
         arg_parser.print_help()
         sys.exit(1)
 
-    sign_cmd = "{} sign --enclave {} {}".format(args.oesign_path, args.enclave_path, args.oesign_args)
-    launch_cmd = "{} {}.signed".format(args.host_path, args.enclave_path)
+    if args.digest_args:
+        digest_cmd = "{} digest --enclave {} {}".format(args.oesign_path, args.enclave_path, args.digest_args)
```

Comment:
It's safer to assemble commands as lists, and to not pass shell=True to subprocess.call(). In particular, it avoids having to deal with paths that contain values that will break tokenisation such as spaces, which this doesn't do correctly.

Reply:
Thanks for the info, I've made the changes but I'll submit it as a PR after this one to improve the python scripts as that probably deserves a separate review.

Reply:
Can we avoid the python scripts by just chaining the sign and run commands?
The idiomatic way to pass arguments would be to use `shlex.join`

Reply:
Strictly speaking, you probably could, but it was a real pain to do to get the bash command line to work correctly with the appropriate characters on both windows and linux. Just for readability purposes, I'd keep the script, for the same reason I'd rather have cmake invoke a .sh rather than copy and paste blobs of concatenated parameters in cmake directly.

Regarding idiomatic invocation of commands, Amaury is correct, the recommendation is to pass lists of args into `subprocess.call()` instead of stringifying them first anyway and the follow-up PR will deal with that.